tools/autograd/derivatives.yaml

```diff
@@ -208,7 +208,7 @@
 
 - name: add.Scalar(Tensor self, Scalar other, Scalar alpha=1) -> Tensor
   self: handle_r_to_c(self.scalar_type(), grad)
-  result: self_t
+  result: self_t.clone()
```

Comment:
This should be done implicitly already by `set_fw_grad()`. As the "viewness" of the tensor and its tangent don't match.
If that is not happening, I would argue that we should fix `set_fw_grad()`.

Reply:
Ho is `self_t` a non-differentiable view here?

Reply:
Doesn't even have to be. `self_t` can just be a plain tensor here. The issue is that the two dual tensors end up sharing the same tangent when the op is supposed to be out-of-place. Unless set_fw_grad or VariableType checks TensorImpl use count, it would not know this. I think the invariant forward AD should be checking here is: that if the forward is out-of-place, then the forward grad should also be out-of-place (similar to the tensorimpl use count checks we have already).

Reply:
From offline discussion: Indeed out of place ops should have out of place formulas.
And since this is a property of the function and not of the Tensor, we do want the function to do this.
We should add a check for these and the `.clone()` is a good solution.